### PR TITLE
boards/arduino-due: added support for Arduino API

### PR DESCRIPTION
**Pull request by haukepetersen**

### CODE REVIEW

File: boards/arduino-due/include/arduino_pinmap.h [hunk 1]
```diff
@@ -30,85 +32,85 @@ extern "C" {
...
-#define ARDUINO_PIN_70          GPIO_PIN(PORT_A, 17)
-#define ARDUINO_PIN_71          GPIO_PIN(PORT_A, 18)
-#define ARDUINO_PIN_72          GPIO_PIN(PORT_C, 30)
-#define ARDUINO_PIN_73          GPIO_PIN(PORT_A, 21)
-#define ARDUINO_PIN_74          GPIO_PIN(PORT_A, 25)
-#define ARDUINO_PIN_75          GPIO_PIN(PORT_A, 26)
-#define ARDUINO_PIN_76          GPIO_PIN(PORT_A, 27)
-#define ARDUINO_PIN_77          GPIO_PIN(PORT_A, 28)
-#define ARDUINO_PIN_78          GPIO_PIN(PORT_B, 23)
+#define ARDUINO_PIN_0           GPIO_PIN(PA, 8)
```
Comment (PeterKietzmann) [inline boards/arduino-due/include/arduino_pinmap.h]: After all this refactoring I'm quit confused actually. Where is `PA` defined?

Comment (haukepetersen) [inline boards/arduino-due/include/arduino_pinmap.h]: `cpu/sam3/include/periph_cpu.h`

Comment (PeterKietzmann) [inline boards/arduino-due/include/arduino_pinmap.h]: Just a side question: Why don't we give similar names on different CPUs? E.g. on stm boards it is `PORT_A`

Comment (PeterKietzmann) [inline boards/arduino-due/include/arduino_pinmap.h]: PS: Once clarified we can hit the green button below :-)

Comment (haukepetersen) [inline boards/arduino-due/include/arduino_pinmap.h]: Simple: so far we are using the naming that is used in a CPU's reference manual/in their vendor headers. So STM is talking about `PORT_A`, Atmel is talking about `PA`, NXP is naming it `P0`. So as these defines are CPU specific anyhow, it makes sense to follow the vendor naming scheme...

**Comment by PeterKietzmann** [comment]
Well, ok. Even if I'm not 100% sure if this is beneficial, it's not the question of this PR. ACK and go

**Comment by OlegHahm** [comment]
To me it seems beneficial.

**Pull request merged**